Code Style Fixes for administrator\components\com_categories

photodude opened the pull request:
Pull Request for Code Style Fixes for administrator\components\com_categories.
#### Summary of Changes
- Multi-line function call not indented correctly
- Line exceeds 150 characters
- Function return type is not void, but function is returning void here
- Block comment text must start on a new line
- Multi-line block comment not used
- Comment must start with a capital letter
- Expected "integer" but found "int" for @var tag in member variable
  comment

[Automatically fixed with PHPCS2 fixers](https://github.com/joomla/coding-standards/pull/109) and some minor edits
#### Testing Instructions

Merge by code review (or just test that categories still function normally in front end and back end)

photodude committed:
Code Style Fixes

- Multi-line function call not indented correctly
- Line exceeds 150 characters
- Function return type is not void, but function is returning void here
- Block comment text must start on a new line
- Multi-line block comment not used
- Comment must start with a capital letter
- Expected "integer" but found "int" for @var tag in member variable
comment

photodude committed:
Fix phpcs 1.x ruleset violations

photodude committed:
another phpcs 1.x ruleset fix

zero-24 commented on administrator/components/com_categories/models/category.php:
```diff
@@ -338,7 +338,9 @@ protected function loadFormData()
 				$data->set('language', $app->input->getString('language', (!empty($filters['language']) ? $filters['language'] : null)));
-				$data->set('access', $app->input->getInt('access', (!empty($filters['access']) ? $filters['access'] : JFactory::getConfig()->get('access'))));
+				$data->set('access',
+							$app->input->getInt('access', (!empty($filters['access']) ? $filters['access'] : JFactory::getConfig()->get('access')))
```
what happend here? maybe we should use:

```
$data->set(
       'access',
       $app->input->getInt('access', (!empty($filters['access']) ? $filters['access'] : JFactory::getConfig()->get('access')))
);
```

zero-24 commented on administrator/components/com_categories/views/category/view.html.php:
```diff
@@ -205,12 +207,17 @@ protected function addToolbar()
 		// Compute the ref_key if it does exist in the component
-		if (!$lang->hasKey($ref_key = strtoupper($component . ($section ? "_$section" : '')) . '_CATEGORY_' . ($isNew ? 'ADD' : 'EDIT') . '_HELP_KEY'))
+		if (!$lang->hasKey(
+			$ref_key = strtoupper($component . ($section ? "_$section" : '')) . '_CATEGORY_' . ($isNew ? 'ADD' : 'EDIT') . '_HELP_KEY'
```
what do you think about 

```
$ref_key = strtoupper($component . ($section ? "_$section" : '')) . '_CATEGORY_' . ($isNew ? 'ADD' : 'EDIT') . '_HELP_KEY';
if (!$lang->hasKey($ref_key))
```

it looks better to me ;)

photodude replied:
Agreed, I debated that change initially. I'll implement it.

zero-24 commented:
Looks good exepted the both comments above. :+1:

wilsonge commented on administrator/components/com_categories/models/category.php:
```diff
@@ -73,7 +73,7 @@ protected function canDelete($record)
 			if ($record->published != -2)
 			{
-				return;
+				return 0;
```
As this is documented a return bool can we return false rather than 0 - i know practically it's the same thing but....

photodude replied:
Good catch, I think our code standard actually requires false or true over 0 or 1. Force of habit from other projects. I'll fix that.

wilsonge commented on administrator/components/com_categories/models/fields/categoryedit.php:
```diff
@@ -251,8 +251,9 @@ protected function getOptions()
-				 * unless the item is already in that category.
+				/*
+				 * To take save or create in a category you need to have create rights for that category
+				 * Unless the item is already in that category.
```
this is a continuation of the previous line so no capital U here for 'unless

photodude replied:
This will also need to be addressed in the draft PHPCS2 ruleset. I'm not sure if the current rule is smart enough to figure out when the comment is a continuation or if it got tripped up by a different fixer.

wilsonge commented on administrator/components/com_categories/models/fields/categoryedit.php:
```diff
@@ -264,8 +265,9 @@ protected function getOptions()
-			 * option to change the category parent for a category or the category for a content item,
+			/*
+			 * If you are only allowed to edit in this category but not edit.state, you should not get any
+			 * Option to change the category parent for a category or the category for a content item,
```
again with the capital at start of line here

wilsonge commented on administrator/components/com_categories/models/fields/categoryparent.php:
```diff
@@ -133,8 +133,9 @@ protected function getOptions()
-				 * unless the item is already in that category.
+				/*
+				 * To take save or create in a category you need to have create rights for that category
+				 * Unless the item is already in that category.
```
Same for the captials on line 2 in this file as well :)

wilsonge commented:
Seems good. We need to fix the return of the `canDelete` thing to `false` otherwise just me being super pedantic

brianteeman commented:
It's good to be pedantic

photodude committed:
Fix items listed in PR comments

- For a return bool we return false rather than 0
- Change some line breaks
- Fix some comment capitalizations

wilsonge commented:
RTC

wilsonge commented:
Bad bot

rdeutz commented:
I have heard pedantic :-)

Now canDelete can result with true, false or nothing (what would be cast to false) but we can change it to:

``` PHP

	protected function canDelete($record)
	{
		if (empty($record->id) || $record->published != -2)
		{
			return false;
		}
			
		return JFactory::getUser()->authorise('core.delete', $record->extension . '.category.' . (int) $record->id);
		
	}
```

would make it more clear what is does

photodude commented:
@wilsonge What do you think about @rdeutz proposed correction to the `canDelete()` function? 

I think it's better than the existing function, But this PR is already RTC.

rdeutz commented:
@photodude you can commit here, then the bot will remove the RTC and I or someone else can add RTC then, I think there isn't so much danger that we break Joomla with that change :-)

photodude committed:
canDelete could return true, false or nothing

Fix `canDelete()` to make the code more clear to the method's actual intent.

brianteeman commented:
The bot didnt remove the label but I have

rdeutz commented:
I think we can set it on RTC the change is so simple that we don't need the whole process

brianteeman commented:
@rdeutz your call. I just did what was asked ;)

rdeutz commented:
at lease @wilsonge has to decide if he will merge it :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11393">issues.joomla.org/joomla-cms/11393</a>.</sub>
